File: kotlin/src/main/kotlin/benchmarks/Base64Encode.kt
package benchmarks

import Benchmark
import java.util.Base64

class Base64Encode : Benchmark() {
    private var n: Long = 0
    private lateinit var bytes: ByteArray
    private lateinit var str2: String
    private var resultVal: UInt = 0u

    init {
        n = configVal("size")
    }

    override fun prepare() {
        val str = "a".repeat(n.toInt())
        bytes = str.toByteArray()
        str2 = Base64.getEncoder().encodeToString(bytes)
    }

    override fun run(iterationId: Int) {
        val encoded = Base64.getEncoder().encodeToString(bytes)
        resultVal += encoded.length.toUInt()  
    }

    override fun checksum(): UInt {
        val str = String(bytes.copyOfRange(0, 4))
        val message = "encode ${str.take(4)}... to ${str2.take(4)}...: $resultVal"
        return Helper.checksum(message)
    }

    override fun name(): String = "Base64Encode"
}

File: scala/src/main/scala/benchmarks/Base64Encode.scala
package benchmarks

import java.util.Base64

class Base64Encode extends Benchmark:
  private var n: Long = 0
  private var str: Array[Byte] = _
  private var str2: String = _
  private var resultVal: Long = 0L

  override def name(): String = "Base64Encode"

  override def prepare(): Unit =
    n = configVal("size")
    val _str = "a" * n.toInt
    str = _str.getBytes
    str2 = Base64.getEncoder.encodeToString(str)

  override def run(iterationId: Int): Unit =
    str2 = Base64.getEncoder.encodeToString(str)
    resultVal += str2.length.toLong

  override def checksum(): Long =
    val _str = new String(str.take(4), "UTF-8")
    val message = s"encode ${_str.take(4)}... to ${str2.take(4)}...: $resultVal"
    Helper.checksum(message)
